crates/flowctl/src/lib.rs

```diff
@@ -139,6 +140,20 @@ impl Cli {
             config_dirty: false,
         };
 
+        // Auth commands are exempt from refreshing auth token, this is to avoid a situation where
+        // an expired refresh_token prevents the user from setting a new service account
+        if !matches!(self.cmd, Command::Auth(_)) {
+            if let Some(api) = context.config.api.as_mut() {
```

Comment:
Very good point, I initially tried to do this in `api_exec` but that was the wrong place since the client was already created. The client creation code is the best place to do this